src/librustc_trans/monomorphize.rs

```diff
@@ -113,34 +114,18 @@ pub fn monomorphic_fn<'a, 'tcx>(ccx: &CrateContext<'a, 'tcx>,
     match map_node {
         hir_map::NodeItem(&hir::Item {
             ref attrs,
-            node: hir::ItemFn(ref decl, _, _, _, _, ref body), ..
+            node: hir::ItemFn(..), ..
         }) |
         hir_map::NodeImplItem(&hir::ImplItem {
             ref attrs, node: hir::ImplItemKind::Method(
-                hir::MethodSig { ref decl, .. }, ref body), ..
+                hir::MethodSig { .. }, _), ..
         }) |
         hir_map::NodeTraitItem(&hir::TraitItem {
             ref attrs, node: hir::MethodTraitItem(
-                hir::MethodSig { ref decl, .. }, Some(ref body)), ..
+                hir::MethodSig { .. }, Some(_)), ..
         }) => {
             attributes::from_fn_attrs(ccx, attrs, lldecl);
-
-            let is_first = !ccx.available_monomorphizations().borrow()
-                                                             .contains(&symbol);
```

Comment:
Yes, the partitioner is now responsible for assigning the correct linkage to monomorphizations.